### DISCUSSION TITLE
How does Wezterm compare to Alacritty

### DISCUSSION THREAD

**Discussion by ghost** [wezterm/wezterm#1769] [General]
What are the differences between these two projects? When should users side with one over the other? Why do we as a Rust community need two different terminals? Do they have different approaches/philosophies?

I think it would be nice not only for me, but also for others to have these questions answered :)

**Comment by CIAvash** [comment]
This question probably should be asked in discussions, since it's not an issue.

You can look at their features:
- Alacritty: https://github.com/alacritty/alacritty/blob/master/docs/features.md
- wezterm: https://wezfurlong.org/wezterm/features.html

In addition to those, wezterm has experimental Bidi/RTL support, workspaces, animations, events, ...

Basically I think wezterm has more features and is more configurable and its developer is more friendly and is more active in adding new features.

I personally think the Rust community has a small part in this(although big part in dependencies and contributing), because it's an application used by all, not a library used by Rust developers. Besides it's always good to have more options to choose from. Every developer has different approaches in both designing and implementing a project and its features.

If someone wants an easy answer, use wezterm or Alacritty if you want a simple GPU-accelerated terminal. Use wezterm if you need the features only wezterm provides. But if someone is using Wayland, then I'd suggest using [foot](https://codeberg.org/dnkl/foot) as a simple, fast and very lightweight terminal.

Also this bit from Alacritty's FAQ may answer some questions:
> Why isn't feature X implemented?
> ... This could be for a number of reasons, but sometimes it's just not a good fit for Alacritty. This means you won't find things like tabs or splits (which are best left to a window manager or terminal multiplexer) nor niceties like a GUI config editor.

About philosophies, only @wez can answer that.

**Comment by wez** [comment]
My perception of Alacritty is:
* Hyper-focused on speed above function, with relatively minimal feature set as a result.
* Their default disposition is to not accept features in order to maintain their performance baseline.

My philosophy/motivation for wezterm is:
* I'm a programmer and I want a programmers terminal
* I use multiple operating systems and I want my terminal to behave consistently across them
* I want it to look good and feel fast for the things that matter most: interactive latency, rather than raw throughput
* I'm sick of terminal emulation mismatches that prevent basic things like italic/underline/true color attributes from working when eg: using remote tmux + mosh, so the multiplexing feature in wezterm exists to eliminate that problem

> Why do we as a Rust community need two different terminals?

The language choice is an implementation detail and the Rust language community itself has no requirements for, or say in, the number of projects that can or should be implemented in Rust.

**Comment by okuma10** [comment]
Just found WezTerm, been using Alacritty for several months.I loved it but there are some features that are missing and some that outright don't work. Ever since I started using it on my Windows machine, Nvim mouse inputs are not there. Seen some people reporting in Alacritty's issue section, and all the responses I saw were "well it's windows, there is nothing to do about this".

**Comment by netherstar** [comment]
My job is working on terminal everydays, for me:

Alacritty: 
- Weird name and hard to remember for me who bad at English
- Weird icon, look like a 80s cyberpunk game icon
- Weird default color scheme
- Dev seem to be not friendly
- No tab, i use tmux all the time but tab still usefull, lack of features
- I don't like it's config file 
- Poor document
- I tried to love it but i can't

Wezterm: 
- Coolname and easy to remember for me from first time
- Simple but cool icon
- Default settings look good and make sense
- More cool dev
- All features i want and i wish
- Easy config with Lua 
- Useful website with all informations i need
- Love Weztem from first time using.

Alacritty is overrated, i don't need a fastest terminal, i need stable and useful features terminal.
Wezterm > Windows Terminal > MobaXterm > Alacritty = Cmder > ConEmu = Zoc =PuTTy

**Reply by kumawatdarshan** [reply]
Alacritty is a coolest name of all the terminals imo.

**Reply by simdimdim** [reply]
it is witty, I give you that, combining alacrity and tty.

**Reply by cjasaMHC** [reply]
based

**Comment by sober-wang** [comment]
I run wezTerm on windwos , The wezTerm process spend 50 MB memory than Windows Terminal more.
sometimes the need more memory ,eg: 200MB or 300MB
![image](https://github.com/wez/wezterm/assets/35804630/4677c6dd-72b5-4add-88a7-e8398615dafe)

**Comment by SalahEddineGhamri** [comment]
My experience nvim is smoother on WezTerm compared to Alacritty on windows machine. Startup is faster and typing is smother. Windows terminal is the worst experience ever. I wished if WezTerm had bash command integration included like Windows terminal.

**Comment by itokun99** [comment]
I use alacritty for several month and I love it, but I just found wezterm after I stuck on alacritty configuration and I love it too, it more than just terminal replacement, wez is programmer terminal who need best terminal with same configuration when cross OS. Alacritty it's too hard to understand for newer comes.

**Comment by okuma10** [comment]
On my side Allacrity has been pretty good in terms of performance. In Nvim it scrolls really fast, while Wezterm lags a bit behind. I would say Wezterm is like 90% of Allacrity's speed on my machine. But I use Allacrity mainly when I have to edit 1000+ lines of code that I need to be somewhat zoomed out. 
That being said the Wezterm window on Windows is really annoying, it is not scaling properly. I suppose because of some windows decorations but that should be taken account of in the code not by the end user...

**Comment by wez** [comment]
Since the OP has deleted their account, I'm going to close this particular discussion.
